docs/SPEC.md

```diff
@@ -3124,7 +3124,101 @@ Contains has the following parameters:
 | set     | array of bool, int, uint, float, string, time | The set of values to search. |
 
 Example: 
-    `contains(value:1, set:[1,2,3])` will return `true`.   
+    `contains(value:1, set:[1,2,3])` will return `true`.
+
+#### Stream/table functions
+
+These functions allow to extract a table from a stream of tables (`tableFind`) and get its schema,
+columns (`getColumn`), and records (`getRecord`).  
+The example below provides an overview of these functions, further information can be found in the paragraphs below.
+
+```
+data = from(bucket:"telegraf/autogen")
+    |> range(start: -5m)
+    |> filter(fn:(r) => r._measurement == "cpu")
+
+// extract the first available table for which "_field" is equal to "usage_idle"
+t = data |> tableFind(fn: (key) => key._field == "usage_idle")
+// this is the same as 'tableFind(tables: data, fn: (key) => key._field == "usage_idle")'
+
+// extract the "_value" column from the table
+values = t |> getColum(column: "_value")
+// the same as 'getColumn(table: t, column: "_value")'
+
+// extract the first record from the table
+r0 = t |> getRecord(idx: 0)
+// the same as 'getRecord(table: t, idx: 0)'
+```
+
+##### TableFind
+
+TableFind extracts the first table in a stream of table whose group key values match a given predicate. If no table
+is found, the function errors.
+
+It has the following parameters:
+
+| Name | Type                  | Description                                                                     |
+| ---- | ----                  | -----------                                                                     |
+| fn   | (key: object) -> bool | Fn is a predicate function. The result is the first table for which fn is true. |
+
+Example:
+
+```
+t = from(bucket:"telegraf/autogen")
+    |> range(start: -5m)
+    |> filter(fn:(r) => r._measurement == "cpu")
+    |> tableFind(fn: (key) => key._field == "usage_idle")
+
+// explore the schema
+firstColLabel = t.schema[0].label
```

Comment:
I see your point. I considered that too!  

This implementation is coherent with how we represent the schema internally though.

One thing I can see in the future is iterating over objects/arrays. One should be able to explore the schema of a table programmatically in Flux. If we decided for objects, one should be able to iterate over objects

```
for label, colMeta in t.schema {
   stdout.Print(label + " - " + colMeta.type)
}
```

Reply:
I think we'll likely cover iteration with a combination of `map` `reduce` and recursion.  furthermore, it shouldn't be hard to define a schema object as an iterable.  

it goes further than this, though.  If we do it this way, we need to define the order for the user.  That order needs to be consistent (sorted by key?) so that they can reliably index the schema to get what they need.  I think we should strongly consider refactoring this to be a map/object instead of an array.

Reply:
I do agree with you @aanthony1243 . It seems also more natural to me. I don't know if we can merge as it is and later apply the change, or directly make this PR include the schema-as-an-object version